File: app/src/main/java/com/tstudioz/iksica/Utils/NetworkMonitor.java
package com.tstudioz.iksica.Utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by etino7 on 12/17/2019.
 */

public class NetworkMonitor {

    private Context context;

    public NetworkMonitor(Context context) {
        this.context = context;
    }

    public boolean isConnected() {

        ConnectivityManager connectivityManager =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();

        return activeNetwork != null &&
                activeNetwork.isConnectedOrConnecting();
    }
}

File: app/src/main/java/com/tstudioz/iksica/Utils/NetworkMonitor.kt
package com.tstudioz.iksica.Utils

import android.content.Context
import android.net.ConnectivityManager

/**
 * Created by etino7 on 12/17/2019.
 */
class NetworkMonitor(private val context: Context) {

    val isConnected: Boolean
        get() {
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetwork = connectivityManager.activeNetworkInfo
            return activeNetwork != null &&
                    activeNetwork.isConnectedOrConnecting
        }

}